From f2c6912286c4bc70567e30e1d9fd37c3f8e9a5c6 Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <xiaowen.liu@freescale.com>
Date: Tue, 21 Jul 2015 17:26:43 +0800
Subject: [PATCH] Revert "clk: divider: Fix best div calculation for
 power-of-two and table dividers"

This reverts commit 70d5a84c0788acadcf90d70c50df3c64dd02e07a.
---
 drivers/clk/clk-divider.c | 37 +------------------------------------
 1 file changed, 1 insertion(+), 36 deletions(-)

diff --git a/drivers/clk/clk-divider.c b/drivers/clk/clk-divider.c
index 4637697..ec22112 100644
--- a/drivers/clk/clk-divider.c
+++ b/drivers/clk/clk-divider.c
@@ -144,37 +144,6 @@ static bool _is_valid_div(struct clk_divider *divider, unsigned int div)
 	return true;
 }
 
-static int _round_up_table(const struct clk_div_table *table, int div)
-{
-	const struct clk_div_table *clkt;
-	int up = _get_table_maxdiv(table);
-
-	for (clkt = table; clkt->div; clkt++) {
-		if (clkt->div == div)
-			return clkt->div;
-		else if (clkt->div < div)
-			continue;
-
-		if ((clkt->div - div) < (up - div))
-			up = clkt->div;
-	}
-
-	return up;
-}
-
-static int _div_round_up(struct clk_divider *divider,
-		unsigned long parent_rate, unsigned long rate)
-{
-	int div = DIV_ROUND_UP(parent_rate, rate);
-
-	if (divider->flags & CLK_DIVIDER_POWER_OF_TWO)
-		div = __roundup_pow_of_two(div);
-	if (divider->table)
-		div = _round_up_table(divider->table, div);
-
-	return div;
-}
-
 static int clk_divider_bestdiv(struct clk_hw *hw, unsigned long rate,
 		unsigned long *best_parent_rate)
 {
@@ -190,7 +159,7 @@ static int clk_divider_bestdiv(struct clk_hw *hw, unsigned long rate,
 
 	if (!(__clk_get_flags(hw->clk) & CLK_SET_RATE_PARENT)) {
 		parent_rate = *best_parent_rate;
-		bestdiv = _div_round_up(divider, parent_rate, rate);
+		bestdiv = DIV_ROUND_UP(parent_rate, rate);
 		bestdiv = bestdiv == 0 ? 1 : bestdiv;
 		bestdiv = bestdiv > maxdiv ? maxdiv : bestdiv;
 		return bestdiv;
@@ -250,10 +219,6 @@ static int clk_divider_set_rate(struct clk_hw *hw, unsigned long rate,
 	u32 val;
 
 	div = DIV_ROUND_UP(parent_rate, rate);
-
-	if (!_is_valid_div(divider, div))
-		return -EINVAL;
-
 	value = _get_val(divider, div);
 
 	if (value > div_mask(divider))
-- 
1.8.0

